goal_mover_topview_pose:
  ros__parameters:
    cmd_topic: /cmd_vel
    pose_topic: /amr_pose
    obstacle_topic: /obstacle_detected

    enable_goal_pose_sub: true
    goal_topic: /goal_pose
    ignore_goal_topic_during_action: true

    action_name: pinky1/actions/move_to_pid

    pose_frame_id: map
    goal_frame_id: map
    yaw_offset_rad: +1.5708

    # linear_P: 0.5
    # linear_I: 0.0
    # linear_D: 0.0
    
    # angular_P: 0.2
    # angular_I: 0.0
    # angular_D: 0.05
    # enable_pid: true

    linear_P: 0.65
    linear_I: 0.00
    linear_D: 0.04
    
    angular_P: 0.4
    angular_I: 0.0
    angular_D: 0.1
    enable_pid: true # 각속도 튜닝끝 이제 선속도 튜닝해야됌.

    # angle_tolerance_deg: 12.0
    # pos_tolerance: 0.03
    # final_yaw_tolerance_deg: 5.0
    angle_tolerance_deg: 5.0
    pos_tolerance: 0.02
    final_yaw_tolerance_deg: 2.0
    enable_final_align: false
    min_angular_error_for_min_speed_deg: 3.0

    max_linear_speed: 0.30
    max_angular_speed: 1.5
    min_linear_speed: 0.06
    min_angular_speed: 0.10
    min_speed_distance_threshold: 0.30

    control_period_sec: 0.02
    pose_timeout_sec: 0.20
    warn_throttle_sec: 2.0
